### PR TITLE
Article added for Singly Linked list

### LINKED ISSUE #72: Content: Singly Linked list
## Description 

Add the article for `Singly Linked list`. The basic layout for the article should be: 

- What is Singly Linked list?
- Explanation for Singly Linked list
- Algorithmic Steps 
- Code (Use C++/Java and Python; At least two languages should be present and pick up code from NeoAlgo with apt credits to author)
- Sample I/O 
- Time & Space Complexity 

This article should be between 500-700 words in length. You should use traditional Markdown to prepare the content and the use of external images is prohibited. You can however create your own images. Run a proper Grammarly Lint check before submitting the PR.

### PR DESCRIPTION
**Pull request by madihamallick**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #72

### CODE REVIEW

File: docs/data-structures/Singly-Linked-list.mdx [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+- Deletion
+    - **Average Case**: O (1)
+    - **Worst Case**: O (1)
+
+- Space Complexity: O (n)
+
+## Credits
+
+- [Sangeeta Mishra](https://github.com/SangeetaMishra) for the [Python Implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/ds/SinglyLinkedList.py)
+- [Vatsal kesarwani](https://github.com/plazzy99) for the [C++ Implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/search/Interpolation_Search.cpp)
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:589]: This implementation is by @sharur7 [here](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/search/Interpolation_Search.cpp)

File: docs/data-structures/Singly-Linked-list.mdx [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+
+### Sample Output
+```
+43
+45
+56
+20
+67
+```
+## Complexity Analysis
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:570]: Give a space after this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 3]
```diff
@@ -0,0 +1,589 @@
+---
+title: Singly Linked list
+slug: /data-structures/Singly-Linked-list
+---
+
+## Introduction
+Singly Linked list is a type of linear Linked list having collection of variable number of data items called nodes. Each node has two fields - one is the **information field** containing the data value of that node, and the other is the **link field** which contains the memory address of the next node. Since here we have only one link field in each node, it is termed as Singly Linked list.
+
+![Capture](https://user-images.githubusercontent.com/70858557/114307192-2bab1300-9afc-11eb-98cd-cd13459394fd.PNG)
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:9]: Did you make this image?

File: docs/data-structures/Singly-Linked-list.mdx [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+cout<<"\t **************************************************"<<endl;
+
+do
+{
+cout<<"\t 1) Insert at Front\t \t 2) Insert at End "<<endl; 
+cout<<"\t 3) Insert at position\t \t 4) Delete at Front"<<endl;
+cout<<"\t 5) Delete at End\t \t 6) Delete at Position"<<endl;
+cout<<"\t 7) Length\t \t\t 8) Display"<<endl;
+cout<<"\t 9) Exit"<<endl;
+cout<<"\t ***************************************************\n ENTER YOUR CHOICE: "<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:129]: Remove the asterisks

File: docs/data-structures/Singly-Linked-list.mdx [hunk 5]
```diff
@@ -0,0 +1,589 @@
...
+//ele is the input data from the user
+int ele;			
+
+
+int main()            
+{
+int ch;
+//Displaying the Features in a tabular manner 
+cout<< "\n\n\t All Basic Features of Singly Linked List"<<endl;                 
+cout<<"\t **************************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:120]: Remove the asterisks

File: docs/data-structures/Singly-Linked-list.mdx [hunk 6]
```diff
@@ -0,0 +1,589 @@
...
+    cout<<"ENTER THE ELEMENT"<<endl;
+    cin>>ele;
+    ptr->data=ele;
+    ptr->next=NULL;
+if(head==NULL)
+ {
+    head=ptr;
+    tail=ptr;
+         cout<<"Inserted Successfully."<<endl;
+         cout<<"\t **************************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:189]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 7]
```diff
@@ -0,0 +1,589 @@
...
+    break;
+    
+    default:
+    cout<<"Invalid Input. Please enter the Number between 1 to 9 "<<endl;
+    
+  }
+}while(ch!=0);
+    return 0;
+}
+//-------------------------------------------------------------
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:176]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 8]
```diff
@@ -0,0 +1,589 @@
...
+    tail=ptr;
+         cout<<"Inserted Successfully."<<endl;
+         cout<<"\t **************************************************"<<endl;
+ }
+ else
+ {
+    tail->next=ptr;
+    tail=ptr;
+          cout<<"Inserted Successfully."<<endl;
+          cout<<"\t **************************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:196]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 9]
```diff
@@ -0,0 +1,589 @@
...
+ }
+ else
+ {
+    tail->next=ptr;
+    tail=ptr;
+          cout<<"Inserted Successfully."<<endl;
+          cout<<"\t **************************************************"<<endl; 
+ }
+}
+//--------------------------------------------------------------
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:199]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 10]
```diff
@@ -0,0 +1,589 @@
...
+    cout<<"ENTER THE ELEMENT"<<endl;
+    cin>>ele;
+    ptr->data=ele;
+if(head==NULL)
+ {
+    head=ptr;
+    tail=ptr;
+    ptr->next=NULL;
+     cout<<"Inserted Successfully."<<endl;
+     cout<<"\t **************************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:212]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 11]
```diff
@@ -0,0 +1,589 @@
...
+ }
+else
+ {
+    ptr->next=head;
+    head=ptr;
+      cout<<"Inserted Successfully."<<endl;
+      cout<<"\t **************************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:219]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 12]
```diff
@@ -0,0 +1,589 @@
...
+
+ }
+}
+
+//-------------------------------------------------------------
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:224]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 13]
```diff
@@ -0,0 +1,589 @@
...
+        temp=head;
+        for(int i=1;i<=loc-1;i++)
+        {
+             prev=temp;
+             temp=temp->next;
+        }
+        ptr->next=temp;
+        prev->next=ptr;
+        cout<<"Inserted Successfully."<<endl;
+        cout<<"\t *************************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:256]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 14]
```diff
@@ -0,0 +1,589 @@
...
+  }
+    
+}
+
+//-------------------------------------------------------------
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:261]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 15]
```diff
@@ -0,0 +1,589 @@
...
+void Delete_At_Front()
+{
+    if(head==NULL)
+    {
+    cout<<"List is Empty"<<endl;
+        cout<<"\t **************************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:267]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 16]
```diff
@@ -0,0 +1,589 @@
...
+    }
+    else
+    {
+        ptr=head;
+        head=ptr->next;
+	   cout<<"Deleted Successfully"<<endl;
+	   cout<<"\t **************************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:274]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 17]
```diff
@@ -0,0 +1,589 @@
...
+    }
+}
+
+//-------------------------------------------------------------
+void Delete_At_End()
+{
+    if(head==NULL)
+    {
+    cout<<"List is Empty"<<endl;
+        cout<<"\t **************************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:286]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 18]
```diff
@@ -0,0 +1,589 @@
...
+    {
+        temp=head;
+        while(temp->next==NULL)
+        {
+            ptr=temp;
+            temp=ptr->next;
+        }
+    ptr->next=NULL;
+    cout<<"Deleted Successfully"<<endl;
+    cout<<"\t **************************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:298]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 19]
```diff
@@ -0,0 +1,589 @@
...
+{
+    
+    int loc;
+    cout<<"ENTER THE LOCATION"<<endl;
+    cin>>loc;
+    //Here Indexing starts from 1
+    if(loc==0)                      
+    {
+        cout<<"0th position doesn't exist."<<endl;
+            cout<<"\t **************************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:315]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 20]
```diff
@@ -0,0 +1,589 @@
...
+
+    }
+    else
+    {
+        // if LinkedList is empty
+        if(head==NULL)              
+        {
+        cout<<"List is Empty"<<endl;
+            cout<<"\t **************************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:324]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 21]
```diff
@@ -0,0 +1,589 @@
...
+            temp=head;
+            struct node*prev;
+            for(int i=1;i<=loc-1;i++)
+            {
+                prev=temp;
+                temp=temp->next;
+            }
+            prev->next=temp->next;
+            cout<<"Deleted Successfully"<<endl;
+            cout<<"\t **************************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:341]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 22]
```diff
@@ -0,0 +1,589 @@
...
+    int i=1;
+    temp=head;
+    cout<<"S.No\t Data\t Pointer"<<endl;
+    while(temp!=NULL)
+    {
+        cout<<i<<"\t"<<temp->data<<"\t"<<temp->next<<endl;
+        temp=temp->next;
+        i++;
+    }
+    cout<<"\t ***********************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:361]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 23]
```diff
@@ -0,0 +1,589 @@
...
+    int i,Count=0;
+    temp=head;
+    while(temp!=NULL)
+    {
+        Count+=1;
+        temp=temp->next;
+        i++;
+    }
+    cout<<"Total no. of Nodes : "<<Count<<endl;
+    cout<<"\t ***********************************************"<<endl;
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:376]: Remove this line

File: docs/data-structures/Singly-Linked-list.mdx [hunk 24]
```diff
@@ -0,0 +1,589 @@
+---
+title: Singly Linked list
+slug: /data-structures/Singly-Linked-list
+---
+
+## Introduction
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:6]: Give a line break after heading

File: docs/data-structures/Singly-Linked-list.mdx [hunk 25]
```diff
@@ -0,0 +1,589 @@
+---
+title: Singly Linked list
+slug: /data-structures/Singly-Linked-list
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:3]: Make the slug all lower case

File: docs/data-structures/Singly-Linked-list.mdx [hunk 26]
```diff
@@ -0,0 +1,589 @@
...
+## Introduction
+Singly Linked list is a type of linear Linked list having collection of variable number of data items called nodes. Each node has two fields - one is the **information field** containing the data value of that node, and the other is the **link field** which contains the memory address of the next node. Since here we have only one link field in each node, it is termed as Singly Linked list.
+
+![Capture](https://user-images.githubusercontent.com/70858557/114307192-2bab1300-9afc-11eb-98cd-cd13459394fd.PNG)
+
+## Why Singly Linked list ?
+
+Arrays can be easily used to store linear data of similar types but the main drawback with array is that the array must be declared to have some fixed size. And due to this predetermined array size at times there is a lot of memory loss. If the array size cannot be determined in advance, the other alternative i.e. Linked list, is used.
+
+**Creating a Linked list:**
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:15]: Use Heading-3

File: docs/data-structures/Singly-Linked-list.mdx [hunk 27]
```diff
@@ -0,0 +1,589 @@
...
+    int info;
+    struct Node *link;
+}*node1, node2;
+```
+
+## Algorithm
+
+### Operations performed on Singly Linked list
+
+**Traversing a Linked list:**
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:31]: Use Heading-4

File: docs/data-structures/Singly-Linked-list.mdx [hunk 28]
```diff
@@ -0,0 +1,589 @@
...
+
+### Operations performed on Singly Linked list
+
+**Traversing a Linked list:** 
+
+- While the pointer containing address of the next node is not NULL
+- Fetch the data part of the node
+- Advance the pointer to the next node
+
+**INSERTING node at the beginning of the Linked list:**
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:37]: Use Heading-4

File: docs/data-structures/Singly-Linked-list.mdx [hunk 29]
```diff
@@ -0,0 +1,589 @@
...
+
+- Create a node
+- Fill data in information field and set the pointer field of the new node as NULL
+- Attach this node to START and mark it as the START node
+
+**INSERTING node at the end of the Linked list:**
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:43]: Use Heading-4

File: docs/data-structures/Singly-Linked-list.mdx [hunk 30]
```diff
@@ -0,0 +1,589 @@
...
+
+- Identify the last node of the Singly Linked list
+- Create a node
+- Fill data in information field and set the pointer field of the new node as NULL
+- Attach this node to END and mark it as the END node
+
+**INSERTING node at a given position of the Linked list:**
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:50]: Use Heading-4

File: docs/data-structures/Singly-Linked-list.mdx [hunk 31]
```diff
@@ -0,0 +1,589 @@
...
+
+- Count the number of data items in the Singly Linked list
+- If the inserting position is 1, insert at the beginning
+- if the inserting position is at the end, insert at the end
+- if it is in the middle, link the newly created node with its left and right node.
+
+**DELETING node from the beginning of the Linked list:**
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:57]: Use Heading-4

File: docs/data-structures/Singly-Linked-list.mdx [hunk 32]
```diff
@@ -0,0 +1,589 @@
...
+
+- A Pointer (Ptr) stores the first nodes address
+- Address of the second node is assigned to start pointer, so now second node becomes the Head node
+- Free the memory occupied by Ptr
+
+**DELETING node from the end of the Linked list:**
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:63]: Use Heading-4

File: docs/data-structures/Singly-Linked-list.mdx [hunk 33]
```diff
@@ -0,0 +1,589 @@
...
+
+- we move to last node by setting present node as ptr and previous one as preptr
+- when we reach the last node i.e. when last node is ptr and second last preptr
+- Make the preptr as the last node and free ptr
+
+**DELETING node from a given position of the Linked list:**
```
Comment (HarshCasper) [inline docs/data-structures/Singly-Linked-list.mdx:69]: Use Heading-4

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/data-structures/Singly-Linked-list.mdx (hunk 3 above)
Comment (madihamallick) [reply docs/data-structures/Singly-Linked-list.mdx:9]: > Did you make this image?

Yes, I made the image.

Comment (HarshCasper) [reply docs/data-structures/Singly-Linked-list.mdx:9]: That's awesome. We have published a lot of articles in past few days. If you would like to make images on them, please feel free to make an issue.

Comment (madihamallick) [reply docs/data-structures/Singly-Linked-list.mdx:9]: > That's awesome. We have published a lot of articles in past few days. If you would like to make images on them, please feel free to make an issue.

I would love to. I will make issues for them soon.

**Comment by madihamallick** [comment]
@HarshCasper I have made the changes. Please review it.

**Review by HarshCasper — APPROVED** [review]

**Comment by HarshCasper** [comment]
Damn good work! Keep the contributions coming :100:

**Pull request merged**